controllers/dnspolicy_status_test.go

```diff
@@ -149,3 +154,59 @@ func TestPropagateRecordConditions(t *testing.T) {
 		})
 	}
 }
+
+func TestDNSPolicyReconciler_calculateStatus(t *testing.T) {
+	type args struct {
+		ctx       context.Context
+		dnsPolicy *v1alpha1.DNSPolicy
+		specErr   error
+	}
+	tests := []struct {
+		name string
+		args args
+		want *v1alpha1.DNSPolicyStatus
+	}{
+		{
+			name: "Enforced status block removed if policy not Accepted. (Regression test)", // https://github.com/Kuadrant/kuadrant-operator/issues/588
+			args: args{
+				dnsPolicy: &v1alpha1.DNSPolicy{
+					Status: v1alpha1.DNSPolicyStatus{
+						Conditions: []metav1.Condition{
+							{
+								Message: "not accepted",
+								Type:    string(gatewayapiv1alpha2.PolicyConditionAccepted),
+								Status:  metav1.ConditionFalse,
+								Reason:  string(gatewayapiv1alpha2.PolicyReasonTargetNotFound),
+							},
+							{
+								Message: "AuthPolicy has been successfully enforced",
```

Comment:
nit:

```suggestion
								Message: "DNSPolicy has been successfully enforced",
```

Reply:
Copy and paste for the win, I will change it.